src/search/mod.rs

```diff
@@ -0,0 +1,149 @@
+use std::collections::BTreeMap;
+
+use rspotify_model::SearchFilter;
+
+/// Builder used to create search query.
+///
+/// Note that when calling the same function multiple time, the filter will be the text from the
+/// last call
+///
+/// This is converted to the query string using into()
+///
+/// Exemple
+/// ```rust
+/// SearchQuery::default()
+///     .any("foo")
+///     .album("bar")
+/// // Filter on album containing "bar" and anything containing "foo"
+/// ```
+///
+/// For more informations on the different filters, look at the [soptify
+/// documentation](https://developer.spotify.com/documentation/web-api/reference/#/operations/search)
+#[derive(Debug, Default)]
+pub struct SearchQuery<'a> {
+    no_filter_query: &'a str,
+    query_map: BTreeMap<SearchFilter, &'a str>,
+}
+
+impl<'a> SearchQuery<'a> {
+    /// Basic filter where the given string can be anything
+    pub fn any(&mut self, str: &'a str) -> &mut Self {
+        self.no_filter_query = str;
```

Comment:
To me, it'd be more intuitive if calling `any` multiple times appended its contents, instead of replace them.